crates/cargo-bsp/src/cargo_communication/request_actor_state.rs

```diff
@@ -41,6 +49,12 @@ pub struct SuiteTaskProgress {
     pub(super) total: i64,
 }
 
+impl CompileState {
+    pub fn increase_compilation_step(&mut self) {
+        self.compilation_step = self.compilation_step.map(|s| s + 1);
```

Comment:
można jakoś tak, ale nie żeby to miało duże znaczenie

```suggestion
        self.compilation_step.as_mut().map(|s| s += 1);
```